image-rs/src/signature/policy/mod.rs

```diff
@@ -29,76 +86,46 @@ pub enum ErrorInfo {
     UnknownScheme,
 }
 
-// Policy defines requirements for considering a signature, or an image, valid.
-// The spec of it is defined in https://github.com/containers/image/blob/main/docs/containers-policy.json.5.md
+/// Policy defines requirements for considering a signature, or an image, valid.
+/// The spec of it is defined in https://github.com/containers/image/blob/main/docs/containers-policy.json.5.md
 #[derive(Deserialize)]
 pub struct Policy {
-    // `default` applies to any image which does not have a matching policy in Transports.
-    // Note that this can happen even if a matching `PolicyTransportScopes` exists in `transports`.
+    /// `default` applies to any image which does not have a matching policy in Transports.
+    /// Note that this can happen even if a matching `PolicyTransportScopes` exists in `transports`.
     default: Vec<PolicyReqType>,
     transports: HashMap<String, PolicyTransportScopes>,
 }
 
 pub type PolicyTransportScopes = HashMap<String, Vec<PolicyReqType>>;
 
 impl Policy {
-    // Returns Ok(()) if the requirement allows running an image.
-    // WARNING: This validates signatures and the manifest, but does not download or validate the
```

Comment:
We might want to keep something like this warning in our replacement method.

Reply:
Added above `check_image_signature()`